### PR TITLE
Add assertion on CEASE line

### PR DESCRIPTION
**Pull request by mwachs5**
**Related issue**: 
Follow-up to https://github.com/chipsalliance/rocket-chip/pull/2419

**Type of change**: other enhancement

**Impact**: API addition (no impact on existing code)

**Development Phase**: implementation

**Release Notes**

Add an assertion that CEASE will not go low once raised until after reset

### CODE REVIEW

**Comment by hcook** [comment]
LGTM as a notional follow-up to #2419, but this isn't rebased on that and FYI they touch literally the same lines of code

**Comment by mwachs5** [comment]
oh i thought #2419 was already merged, my bad

**Comment by aswaterman** [comment]
Thanks @mwachs5, #2419 is merged now, so you can rebase

File: src/main/scala/tile/Interrupts.scala [hunk 1]
```diff
@@ -102,7 +102,13 @@ trait SourcesExternalNotifications { this: BaseTile =>
       saturated
     }
     val (cease, _) = ceaseNode.out(0)
-    cease(0) := could_cease.map(waitForQuiescence(_)).getOrElse(false.B)
+    cease(0) := could_cease.map{ c => 
+      val cease = (waitForQuiescence(c))
+      // Test-Only Code --
+      val prev_cease = RegNext(c, false.B)
+      assert((!(prev_cease & !c) | reset.asBool), "CEASE line can not glitch once raised")
```
Comment (mwachs5) [inline src/main/scala/tile/Interrupts.scala:109]:
```suggestion
      assert(!(prev_cease & !c), "CEASE line can not glitch once raised") 
```

I *think* assertions are already disabled during reset? @jchang0 ?

**Comment by aswaterman** [comment]
chisel assertions indeed are disabled during reset - it’s literally
implemented as assert(reset || cond), IIRC

**Comment by mwachs5** [comment]
@jchang0 can I get a review on this?

**Pull request merged**